src/java.base/share/classes/java/text/PatternEntry.java

```diff
@@ -287,8 +287,8 @@ public PatternEntry next() throws ParseException {
         }
 
         // We re-use these objects in order to improve performance
-        private StringBuffer newChars = new StringBuffer();
-        private StringBuffer newExtension = new StringBuffer();
+        private StringBuilder newChars = new StringBuilder();
+        private StringBuilder newExtension = new StringBuilder();
```

Comment:
Again, in 2022, I don't know that this cache+reuse pattern is needed for performance.

Reply:
Due to some [bootstrapping issues](https://stackoverflow.com/questions/71834059/why-invokedynamic-based-string-concatenation-is-not-available-for-javac-in-java) in `java.base` explicit String concatenation is not translated into invokedynamic, so in practice it's faster to use `StringBuilder.append()` than `str1 + str2`, see e.g. https://github.com/openjdk/jdk/pull/3903/files

Reply:
Thanks. I was not suggesting that we use string concatenation, but rather just remove the `newChar`/`newExtension` fields, and replace, e.g.:
`214             newChars.setLength(0);`
with
`214            StringBuilder newChars = new StringBuilder();`
though I'll admit that I don't have a good sense of whether that's worth doing.

Reply:
Memory you don't allocate doesn't need to be GC'd.  I'd leave it alone and save the GC overhead.